lang/src/bpf_writer.rs

```diff
@@ -26,6 +26,17 @@ impl Write for BpfWriter<&mut [u8]> {
         );
         sol_memcpy(&mut self.inner[(self.pos as usize)..], buf, amt);
         self.pos += amt as u64;
+        // Set remaining bytes to 0
+        if pos < self.inner.len() {
+            let remaining_bytes = self.inner.len() - pos;
+            sol_memcpy(
+                &mut self.inner[pos..],
+                vec![0; remaining_bytes].as_ref(),
```

Comment:
`BpfWriter` is a essentially a simple cursor implementation and it's expected for the `write` method to only write `buf.len()` amount of bytes and not touch the other parts.

Reply:
Thank! This is not the correct way to do it I have a better one. I can push that once we know this is required in anchor 😄. I also changed the implementation in our smart contract.